### PR TITLE
Github Actions: Add Phpstan

### PR DESCRIPTION
**Pull request by sukhwinder33445**

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -40,3 +46,7 @@ jobs:
       - name: PHP CodeSniffer
         if: success() || matrix.allow_failure
         run: phpcs -wps --colors
+
+      - name: PHPStan
+        uses: php-actions/phpstan@v3
+        if: always()
```
Comment (sukhwinder33445) [inline .github/workflows/php.yml:52]: Since we no longer use Psalm, please set this back to: `if: success() || matrix.allow_failure`.

File: application/controllers/ReportController.php [hunk 2]
```diff
@@ -192,8 +197,9 @@ public function downloadAction()
 
         switch ($type) {
             case 'pdf':
-                /** @var Hook\PdfexportHook */
-                Pdfexport::first()->streamPdfFromHtml($this->report->toPdf(), $name);
+                /** @var Hook\PdfexportHook $exports */
+                $exports = Pdfexport::first();
+                $exports->streamPdfFromHtml((string) $this->report->toPdf(), $name);
```
Comment (sukhwinder33445) [inline application/controllers/ReportController.php:202]: `PdfExport::streamPdfFromHtml()` expects `string|PrintableHtmlDocument`.

File: library/Reporting/Mail.php [hunk 3]
```diff
@@ -33,7 +33,7 @@ class Mail
      */
     public function getFrom()
     {
-        if (isset($this->from)) {
+        if (! empty($this->from)) {
```
Comment (sukhwinder33445) [inline library/Reporting/Mail.php:36]: The check for null is sufficient.

File: library/Reporting/Web/Forms/ScheduleForm.php [hunk 4]
```diff
@@ -35,7 +37,9 @@ class ScheduleForm extends CompatForm
     public function __construct()
     {
         $this->scheduleElement = new ScheduleElement('schedule_element');
-        $this->scheduleElement->setIdProtector([Icinga::app()->getRequest(), 'protectId']);
+        /** @var EmbeddedWeb $app */
```
Comment (sukhwinder33445) [inline library/Reporting/Web/Forms/ScheduleForm.php:40]: Please use `Web` instead.

File: phpstan.neon [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    ignoreErrors:
+        -
+            messages:
+                - '#Unsafe usage of new static\(\)#'
+                - '#. but return statement is missing#'
+            reportUnmatched: false
+
+        - '#Call to an undefined method Icinga\\Module\\Reporting\\RetryConnection::lastInsertId\(\)#'
+
+    universalObjectCratesClasses: # to ignore magic property errors (db columns)
```
Comment (sukhwinder33445) [inline phpstan.neon:27]: Please remove the comment.

File: application/controllers/ReportController.php (hunk 2 above)
Comment (yhabteab) [reply application/controllers/ReportController.php:202]: ?

Comment (sukhwinder33445) [reply application/controllers/ReportController.php:202]: As discussed offline, the abstract class needs to be fixed. But still `PdfExport::streamPdfFromHtml()` also checks if $html is a `PrintableHtmlDocument` instance. In your case it is now always false.

File: library/Reporting/Mail.php (hunk 3 above)
Comment (sukhwinder33445) [reply library/Reporting/Mail.php:36]: I see.

**Review by sukhwinder33445 — COMMENTED** [review]
LGTM 🎉

**Review by yhabteab — APPROVED** [review]

File: library/Reporting/Mail.php [hunk 6]
```diff
@@ -33,7 +33,7 @@ class Mail
      */
     public function getFrom()
     {
-        if (isset($this->from)) {
+        if ($this->from !== null) {
```
Comment (nilmerg) [inline library/Reporting/Mail.php:36]: why is this necessary?

Comment (nilmerg) [reply library/Reporting/Mail.php:36]: @sukhwinder33445 told me already. Please adjust the doctype of this property and $transport instead.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Reporting/Mail.php (hunk 6 above)
Comment (yhabteab) [reply library/Reporting/Mail.php:36]: No! These properties aren't allowed to be null and never to be used with null values. This check only ensures that this property is not yet set and therefore must be initialized properly.

**Review by nilmerg — APPROVED** [review]
